src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,261 @@
+            }
+            try
+            {
+                blobSegment = await container.ListBlobsSegmentedAsync(prefix: null, useFlatBlobListing: true, 
+                    blobListingDetails: BlobListingDetails.None, maxResults: blobPollLimitPerContainer, currentToken: continuationToken, 
+                    options: null, operationContext: null, cancellationToken: cancellationToken);
+                currentBlobs = blobSegment.Results;
+            }
+            catch (StorageException exception)
+            {
+                if (exception.IsNotFound())
+                {
+                    return Enumerable.Empty<IStorageBlob>();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            List<IStorageBlob> newBlobs = new List<IStorageBlob>();
+
+            // Type cast to IStorageBlob is safe due to useFlatBlobListing: true above.
+            foreach (IStorageBlob currentBlob in currentBlobs)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                IStorageBlobProperties properties = currentBlob.Properties;
+                DateTime lastModifiedTimestamp = properties.LastModified.Value.UtcDateTime;
+
+                if (lastModifiedTimestamp > _lastSweepCycleStartTime[container])
+                {
+                    newBlobs.Add(currentBlob);
+                }
+            }
+
+            // record continuation token for next chunk retrieval
+            _continuationTokens[container] = blobSegment.ContinuationToken;
+
+            // if ending a cycle then copy currentSweepCycleStartTime to lastSweepCycleStartTime
+            if (blobSegment.ContinuationToken == null)
+            {
+                _lastSweepCycleStartTime[container] = _currentSweepCycleStartTime[container];
+            }
+
+            return newBlobs;
+        }
+
+        private async Task NotifyRegistrationsAsync(IStorageBlob blob, ICollection<IStorageBlob> failedNotifications,
+            CancellationToken cancellationToken)
+        {
+            IStorageBlobContainer container = blob.Container;
+            ICollection<ITriggerExecutor<IStorageBlob>> registrations;
+
+            // Blob written notifications are host-wide, so filter out things that aren't in the container list.
+            if (!_registrations.TryGetValue(container, out registrations))
+            {
+                return;
+            }
+
+            foreach (ITriggerExecutor<IStorageBlob> registration in registrations)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                FunctionResult result = await registration.ExecuteAsync(blob, cancellationToken);
+                if (!result.Succeeded)
+                {
+                    // If notification failed, try again on the next iteration.
+                    failedNotifications.Add(blob);
+                }
```

Comment:
what does this mean? 
If the user's function throws an exception, does this mean we'll re-execute it?